packages/base/src/Boot.ts

```diff
@@ -7,56 +8,55 @@ import { registerCurrentRuntime } from "./Runtimes.js";
 import { getFeature } from "./FeaturesRegistry.js";
 import type OpenUI5Support from "./features/OpenUI5Support.js";
 import type F6Navigation from "./features/F6Navigation.js";
-import { PromiseResolve } from "./types.js";
 
-let bootPromise: Promise<void>;
+const eventProvider = new EventProvider<void, void>();
+let booted = false;
 
 /**
- * Attach  a callback that will be executed on boot
+ * Attaches a callback that will be executed after boot finishes.
  * @public
- * @param listener
+ * @param { Function } listener
  */
-const attachBoot = async (listener: () => void) => {
-	await boot();
+const attachBoot = (listener: () => void) => {
+	if (!booted) {
+		eventProvider.attachEvent("boot", listener);
+		return;
+	}
+
 	listener();
 };
 
 const boot = async (): Promise<void> => {
-	if (bootPromise !== undefined) {
-		return bootPromise;
+	if (booted) {
```

Comment:
the `bootPromise` needs to stay. In case multiple components call boot, they all need to wait for the same promise to be resolved. otherwise, the first component will trigger a real async boot and the rest of the components will repeat the initialization work multiple times. This is inefficient, there is no need to execute the boot logic multiple times. It could even lead to bugs like the `boot` event fired multiple times or the font face registered multiple times.

Reply:
You are right, the promise must be reused